ostd/src/task/scheduler/mod.rs

```diff
@@ -35,52 +96,246 @@ pub fn inject_scheduler(scheduler: &'static dyn Scheduler<Task>) {
+///   which informs the scheduler that the task is exiting and will never be enqueued again.
+///
+/// The general workflow for OSTD to handle a scheduling event is as follows:
+/// 1. Acquire exclusive access to the local runqueue using [`Scheduler::mut_local_rq_with`].
+/// 2. Call [`LocalRunQueue::update_current`] to update the current task's state,
+///    returning a boolean value that indicates
+///    whether the current task should and can be replaced with another runnable task.
+/// 3. If the task is about to sleep or exit, call [`LocalRunQueue::dequeue_current`]
+///    to remove it from the runqueue.
+/// 4. If the return value of `update_current` in Step 2 is true,
+///    then select the next task to run with [`LocalRunQueue::pick_next`].
+///
+/// ## When to Pick the Next Task?
 ///
-/// This local view provides the interface for the runqueue of a CPU core
-/// to be inspected and manipulated by the code running on this particular CPU core.
+/// As shown above,
+/// OSTD guarantees that `pick_next` is only called
+/// when the current task should and can be replaced.
+/// This avoids unnecessary invocations and improves efficiency.
 ///
-/// Conceptually, a local runqueue consists of two parts:
-/// (1) a priority queue of runnable tasks;
-/// (2) the current running task.
-/// The exact definition of "priority" is left for the concrete implementation to decide.
+/// But under what conditions should the current task be replaced?
+/// Two criteria must be met:
+/// 1. There exists at least one other runnable task in the runqueue.
+/// 2. That task should preempt the current one, if present.
+///
+/// Some implications of these rules:
+/// - If the runqueue is empty, `update_current` must return `false`—there's nothing to run.
+/// - If the runqueue is non-empty but the current task is absent,
+///   `update_current` should return `true`—anything is better than nothing.
+/// - If the runqueue is non-empty and the flag is `UpdateFlags::WAIT`,
+///   `update_current` should also return `true`,
+///   because the current task is about to block.
+/// - In other cases, the return value depends on the scheduler's prioritization policy.
+///   For instance, a real-time task may only be preempted by a higher-priority task
+///   or if it explicitly yields.
+///   A normal task under Linux's CFS may be preempted by a task with smaller vruntime,
+///   but never by the idle task.
+///
+/// When OSTD is unsure about whether the current task should or can be replaced,
+/// it will inovke [`LocalRunQueue::try_pick_next`], the fallible version of `pick_next`.
+///
+/// ## Intern Working
+///
+/// To guide scheduler implementers,
+/// we provide a simplified view of how OSTD interacts with local runqueues _internally_
+/// in order to handle the four scheduling events.
+///
+/// ### Yielding
+///
+/// ```
+/// # use ostd::prelude::*;
+/// # use ostd::task::scheduler::*;
```

Comment:
This isn't enough to satisfy the compiler. We'll need to provide dummy implementations for `switch_to`, `cpu_local::set_need_preempt`, and `cpu_local::need_preempt` to enable the compiler to check whether these examples can be compiled without errors.

However, since we have not yet enabled the check in CI, I'm happy to defer this to future work.

Reply:
How about now?

```rust
/// # use ostd::prelude::*;
/// # use ostd::task::::*;
/// # use ostd::task::scheduler::*;
/// # fn switch_to(next: Arc<Task>) { }
```

Reply:
But `cpu_local` is `ostd::task::preempt::cpu_local`. Meanwhile, `preempt`/`cpu_local` are NOT public modules?

So will adding `ostd::task::*` make any sense?